src/rust/engine/serverset/src/lib.rs

```diff
@@ -220,12 +219,14 @@ impl<T: Clone + Send + Sync + 'static> Serverset<T> {
   ///
   pub fn next(&self) -> BoxFuture<(T, HealthReportToken), String> {
     let now = Instant::now();
-    let server_count = self.inner.connections.len();
+    let mut inner = self.inner.lock();
+    let server_count = inner.connections.len();
 
     let mut earliest_future = None;
     for _ in 0..server_count {
-      let i = self.inner.next.fetch_add(1, Ordering::Relaxed) % server_count;
-      let server = &self.inner.connections[i];
+      let i = inner.next % server_count;
+      inner.next = inner.next.wrapping_add(1);
```

Comment:
Why is this important to `wrapping_add` here? Should we `wrapping_add` everywhere, or is it just because the prior code used modular arithmetic anyway and we want to preserve functionality?

Also, now that I think of it, doesn't the prior code add before taking the index? Not sure it matters a lot, but in the previous code `i = (inner.next++) % server_count`, where now `i = inner.next % server_count; inner.next++`.

Reply:
In practice, this gets incremented once per request, so it's unlikely but not impossible that a pants invocation will overflow this. If we don't `wrapping_add`, we will panic when it overflows (at least, we will in debug mode).

Before we were using `AtomicUsize.fetch_add` which is specified to do a wrapping addition on overflow: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#method.fetch_add - and `fetch_add` returns the value _before_ the addition :) These are great additional reasons to avoid using the weird/unfamiliar `AtomicUsize`!